ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp

```diff
@@ -53,28 +54,46 @@ std::unique_ptr<Process> createComponentTransportProcess(
 
     std::vector<std::vector<std::reference_wrapper<ProcessVariable>>>
         process_variables;
+
+    // Collect all process variables in a vector before allocation
+    // pressure first, concentration then
+    auto const collected_process_variables = findProcessVariables(
+        variables, pv_config,
+        {//! \ogs_file_param_special{prj__processes__process__ComponentTransport__process_variables__pressure}
+         "pressure",
+         //! \ogs_file_param_special{prj__processes__process__ComponentTransport__process_variables__concentration}
+         "concentration"});
+
+    // Check number of components for each process variable
+    auto it = std::find_if(
+        collected_process_variables.cbegin(),
+        collected_process_variables.cend(),
+        [](std::reference_wrapper<ProcessLib::ProcessVariable> const& pv) {
+            return pv.get().getNumberOfComponents() != 1;
+        });
+
+    if (it != collected_process_variables.end())
+        OGS_FATAL(
+            "Number of components for process variable \"%s\" should be 1 "
```

Comment:
Please, use ' instead of \\" for consistency with other message.

Reply:
Fixed.